Backend/routes/history.js

```diff
@@ -51,8 +51,84 @@ router.get('/:userId', async (req, res) => {
         res.json(history);
 
     }
+})
+
+router.get('/registered_days/:userId', async (req, res) => {
+    const userId = req.params.userId;
+    const year = req.query.year;
+    const month = req.query.month;
+    const day = req.query.day;
 
+    if (year && !(month)) {
+        const history = await prisma.meals.groupBy({
+            by: ['date'],
+            where: {
+                userId,
+                date: {
+                    gte: new Date(year + "-01-01"),
```

Comment:
nit: we could creat a helper function to make it ez to read and also remove the need to think about how to format the date (i.e. getting the order of year, month, date right and using hyphens). It could look something like this as you have it on line 98
```
function formatDate(year, month, day) {
   return (year + '-' + month + '-' + day);
}
```